pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -172,14 +235,16 @@ func (l *podNICImpl) sortIPsBasedOnPrimaryIP(ipv4, ipv6 string) ([]string, error
 	return []string{ipv6, ipv4}, nil
 }
 
-func (l *podNICImpl) PlugPhase1(vmi *v1.VirtualMachineInstance, iface *v1.Interface, network *v1.Network, podInterfaceName string, pid int) error {
+func (l *podNIC) PlugPhase1(pid int) error {
+
+	l.launcherPID = &pid
```

Comment:
I honestly do not like mutating the struct here (you do have this value when you create it...), but it really does make everything a lot simpler .

Reply:
What do you think about something like

```golang
func (n *VMNetworking) SetupPodNetworkPhase1(pid int) error {                   
    nics, err := n.getNICsWithLauncherPID(pid)                                                    
    if err != nil {                                                             
        return nil                                                              
    }                                                                           
    for _, nic := range nics {                                                  
        err = nic.PlugPhase1()                                               
        if err != nil {                                                         
            return err                                                          
        }                                                                       
    }                                                                           
    return nil                                                                  
}          
```

Reply:
I would prefer 2 implementations of the VMNetworking interface (that would be a step forward removing this whole concept of phases), but this seems a good compromise. 

I guess that's better than mutating the struct.

Reply:
I like the idea of two implementations of the same interface using a pair of factory functions.

```golang
NewVMNetworkingForLauncherPID(pid).SetupPodNetwork()
NewVMNetworkingForDomain(domain).SetupPodNetwork()
```

Reply:
Exactly. 

I just think the names could be improved according to what they do, not what their input is; something like:

```golang
NewVMNetworkingInfraConfigurator(pid).Setup()
NewVMNetworkingSpecGenerator(domain).Setup()
```

I don't think you have to do this now, we can always do it later on, once we manage to split the bind mechanism into multiple interfaces.

Reply:
fair enough let's go with `getNICsWithLauncherPID` and address it properly at follow up.

Reply:
Done